test/agent_test.go

```diff
@@ -117,6 +117,36 @@ func TestAgentNamespaces(t *testing.T) {
 	require.NotEqual(t, job1.Id, job2.Id)
 }
 
+func TestAgentMultiNode(t *testing.T) {
+	_, _, finish := setupMultiNodeTest("TestAgentMultiNode")
+	defer finish()
+
+	ac1, err := newAgentClient(agentToken(), defaultServerPort)
+	require.NoError(t, err)
+	ac2, err := newAgentClient(agentToken(), defaultServerPort) 
+	defer ac1.close()
+	defer ac2.close()
+	ac1.Run(livekit.JobType_JT_ROOM, "namespace")
+	ac2.Run(livekit.JobType_JT_PUBLISHER, "namespace")
+
+	time.Sleep(time.Second * 3)
+
+	require.Equal(t, int32(1), ac1.registered.Load())
+	require.Equal(t, int32(1), ac2.registered.Load())
+
+	c1 := createRTCClient("c1", secondServerPort, nil) // Create a room on the second node
+	waitUntilConnected(t, c1)
+
+	t1, err := c1.AddStaticTrack("audio/opus", "audio", "micro")
+	require.NoError(t, err)
+	defer t1.Stop()
+
+	time.Sleep(time.Second * 10)
```

Comment:
that's not expected.. but perhaps it's due to the way we are waiting for namespaces to load.

another thing in tests, it's better to use [WithTimeout](https://github.com/livekit/livekit/blob/d87f8aa299a3d9b27dde14fcb2d586683b431084/pkg/testutils/timeout.go#L27) rather than hard sleeps.